metrics/sort.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2023 Palantir Technologies. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !go1.21
+
+package metrics
+
+import (
+	"sort"
+)
+
+// sortStrings is the default sort.Strings function.
+// Unfortunately this forces the slice to escape to the heap.
+// See https://github.com/golang/go/issues/17332
```

Comment:
I didn't feel like publicly linking to the github issue, but this describes the long-time problem.